pkg/resources/syslogng/statefulset.go

```diff
@@ -260,15 +262,26 @@ func (r *Reconciler) bufferMetricsSidecarContainer() *corev1.Container {
 			port = r.Logging.Spec.SyslogNGSpec.BufferVolumeMetrics.Port
 		}
 		portParam := fmt.Sprintf("--web.listen-address=:%d", port)
-		args := []string{portParam, "--collector.disable-defaults", "--collector.filesystem"}
+		args := []string{portParam, "--collector.disable-defaults", "--collector.filesystem", "--collector.textfile", "--collector.textfile.directory=/prometheus/node_exporter/textfile_collector/"}
+
+		nodeExporterCmd := fmt.Sprintf("nodeexporter -> ./bin/node_exporter %v", strings.Join(args, " "))
+		bufferSizeCmd := "buffersize -> /prometheus/buffer-size.sh"
 
-		customRunner := fmt.Sprintf("./bin/node_exporter %v", strings.Join(args, " "))
 		return &corev1.Container{
 			Name:            "buffer-metrics-sidecar",
 			Image:           v1beta1.RepositoryWithTag(bufferVolumeImageRepository, bufferVolumeImageTag),
 			ImagePullPolicy: corev1.PullIfNotPresent,
-			Args:            []string{"--port", "7358", "--startup", customRunner},
-			Ports:           generatePortsBufferVolumeMetrics(r.Logging.Spec.SyslogNGSpec),
+			Args: []string{
+				"--exec", nodeExporterCmd,
```

Comment:
```suggestion
			Args: []string{
			        "--port", "7358",
				"--exec", nodeExporterCmd,
```

This is required, because otherwise the default port (7357) is not going to be already used by the config reloader. Could you please add it back?